schematic/models/validate_attribute.py

```diff
@@ -419,23 +423,26 @@ def get_message_level(
         col_is_recommended = rule_name == "recommended"
         col_is_required = dmge.get_node_required(node_display_name=error_col)
         rules_include_na_modifier = rule_in_rule_list("IsNA", validation_rule_list)
-        error_val_is_na = (
-            error_val.lower() in not_applicable_strings
-            if isinstance(error_val, str)
-            else False
-        )
-
-        if is_schema_error:
-            return "error"
 
         if specified_level:
             return specified_level
 
-        if (
-            error_val_is_na and rules_include_na_modifier
-        ) or rule_name.lower() == "isna":
+        if rule_name.lower() == "isna":
+            return None
```

Comment:
I would suggest moving these two checks (lines 427 through 431) closer to the top of the method as you could avoid some uneeded computation and possibly speed it up.